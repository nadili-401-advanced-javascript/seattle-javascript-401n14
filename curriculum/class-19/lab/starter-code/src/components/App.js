import React from 'react';
import Converter from './Converter';

function App() {
    return (
        <div className='App'>
            <h1>Multi-Purpose Converter</h1>
            <Converter xLabel='Miles' yLabel='Kilometers' rate={1.61} />
            <Converter xLabel='Cups' yLabel='Fluid Ounces' rate={8} />
            <Converter xLabel='Inches' yLabel='Centimeters' rate={2.54} />
        </div>
    );
}

export default App;
