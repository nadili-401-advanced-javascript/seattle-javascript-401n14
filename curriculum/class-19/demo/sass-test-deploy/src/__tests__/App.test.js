import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../components/App';
configure({ adapter: new Adapter() });

describe('App component', () => {
    it('renders correctly', () => {
        const page = renderer.create(<App />).toJSON();
        expect(page).toMatchSnapshot();
    });

    it('has a Counter', () => {
        let component = mount(<App />);
        expect(component.find('div.bordered-box h2').text()).toBe(
            'Your current count is: 0',
        );
    });

    it('adds one to Counter', () => {
        let component = mount(<App />);
        let addBtn = component.find('div.bordered-box button#add');

        addBtn.simulate('click');
        expect(component.find('div.bordered-box h2').text()).toBe(
            'Your current count is: 1',
        );
    });
});
