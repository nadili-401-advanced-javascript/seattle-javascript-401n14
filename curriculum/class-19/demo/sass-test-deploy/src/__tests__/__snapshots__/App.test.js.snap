// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="container App"
>
  <h1>
    Demo Application
  </h1>
  <div
    className="bordered-box"
  >
    <h2>
      Your current count is: 
      <span>
        0
      </span>
    </h2>
    <button
      id="subtract"
      onClick={[Function]}
    >
      Subtract
    </button>
    <button
      id="add"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
